Assets/Spawner/Spawner.cs

```diff
@@ -35,14 +40,24 @@ private void Awake() {
 
   void Start() {
     if (filename.Length == 0) return;
-    Waves? waves = Deserialize<Waves>(filename);
+    waves = Deserialize<Waves>(filename);
     if (waves != null) {
       ObjectPool.Instance.InitializeObjectPool(waves.GetEnemyKeys());
       SpawnWaves(waves);
     } else {
       // TODO: Make this an error.
       Debug.Log("ERROR: Waves is null.");
     }
+    // Send initial timing data to EnemySpawnIndiciatorManager.
+    EnemySpawnIndicatorManager.Instance.EnemySpawnTimes = GetSpawnTimes();
```

Comment:
How about the UpdateSpawnIndicatorData has type of EnemySpawn times and just passes that along.
The EnemySpawnIndicatorManager is then decoupled from this class.

Reply:
This has been addressed. The coupling has been removed.